contracts/okp4-cognitarium/src/state/storer.rs

```diff
@@ -0,0 +1,216 @@
+                    triple.predicate.clone(),
+                    triple.subject.clone(),
+                ),
+                &triple,
+            )
+            .map_err(ContractError::Std)
+    }
+
+    pub fn finish(&mut self) -> Result<Uint128, ContractError> {
+        STORE.save(self.storage, &self.store)?;
+        NAMESPACE_KEY_INCREMENT.save(self.storage, &self.ns_key_inc_offset)?;
+        for entry in &self.ns_cache {
+            namespaces().save(self.storage, entry.0.to_string(), entry.1)?;
+        }
+
+        Ok(self.store.stat.triple_count - self.initial_triple_count)
+    }
+
+    fn resolve_namespace_key(&mut self, ns_str: String) -> StdResult<u128> {
+        match self.ns_cache.get_mut(ns_str.as_str()) {
+            Some(namespace) => {
+                namespace.counter += 1;
+                Ok(namespace.key)
+            }
+            None => {
+                let mut namespace = match namespaces().load(self.storage, ns_str.clone()) {
+                    Err(StdError::NotFound { .. }) => {
+                        let n = Namespace {
+                            key: self.ns_key_inc_offset,
+                            counter: 0u128,
+                        };
+                        self.ns_key_inc_offset += 1;
+                        Ok(n)
+                    }
+                    Ok(n) => Ok(n),
+                    Err(e) => Err(e),
+                }?;
+
+                namespace.counter += 1;
+                self.ns_cache.insert(ns_str, namespace.clone());
+                Ok(namespace.key)
+            }
+        }
+    }
+
+    fn rio_to_triple(&mut self, triple: model::Triple) -> StdResult<Triple> {
+        Ok(Triple {
+            subject: self.rio_to_subject(triple.subject)?,
+            predicate: self.rio_to_node(triple.predicate)?,
+            object: self.rio_to_object(triple.object)?,
+        })
+    }
+
+    fn rio_to_subject(&mut self, subject: model::Subject) -> StdResult<Subject> {
+        match subject {
+            model::Subject::NamedNode(node) => self.rio_to_node(node).map(Subject::Named),
+            model::Subject::BlankNode(node) => Ok(Subject::Blank(node.id.to_string())),
+            _ => Err(StdError::generic_err("RDF star syntax unsupported")),
+        }
+    }
+
+    fn rio_to_node(&mut self, node: model::NamedNode) -> StdResult<Node> {
+        let (ns, v) = rdf::explode_iri(node.iri)?;
+        Ok(Node {
+            namespace: self.resolve_namespace_key(ns)?,
+            value: v,
+        })
+    }
+
+    fn rio_to_object(&mut self, object: model::Term) -> StdResult<Object> {
+        match object {
+            Term::BlankNode(node) => Ok(Object::Blank(node.id.to_string())),
+            Term::NamedNode(node) => self.rio_to_node(node).map(Object::Named),
+            Term::Literal(literal) => self.rio_to_literal(literal).map(Object::Literal),
+            _ => Err(StdError::generic_err("RDF star syntax unsupported")),
```

Comment:
Certainly it will never occurs but in case a new enum case is added, the build will tell us that we need to handle this new case :).

```suggestion
            model::Subject::Triple(_) => Err(StdError::generic_err("RDF star syntax unsupported")),
```

Reply:
Make sense! I'll change that 😉